src/api.v2/helpers/pipeline/subset.js

```diff
@@ -0,0 +1,66 @@
+const sqlClient = require('../../../sql/sqlClient');
+const getLogger = require('../../../utils/getLogger');
+const { GEM2S_PROCESS_NAME } = require('../../constants');
+const { createSubsetPipeline } = require('./pipelineConstruct');
+const Experiment = require('../../model/Experiment');
+const ExperimentExecution = require('../../model/ExperimentExecution');
+const ExperimentParent = require('../../model/ExperimentParent');
+const UserAccess = require('../../model/UserAccess');
+
+const logger = getLogger('[SubsetController] - ');
+
+const createExperimentToSubset = async (parentExperimentId, userId, name) => {
+  logger.log(`Creating experiment to subset ${parentExperimentId}`);
+  let subsetExperimentId;
+
+  await sqlClient.get().transaction(async (trx) => {
+    subsetExperimentId = await new Experiment(trx).createCopy(parentExperimentId, name);
+    await new UserAccess(trx).createNewExperimentPermissions(userId, subsetExperimentId);
+
+    await new ExperimentParent(trx).create(
+      { experiment_id: subsetExperimentId, parent_experiment_id: parentExperimentId },
+    );
+  });
+
+  // Samples are not created here, we add them in handleResponse of SubsetSeurat
+  logger.log(`Created ${subsetExperimentId}, subsetting experiment ${parentExperimentId} to it`);
+
+  return subsetExperimentId;
+};
+
+const executeSubsetPipeline = async (params, authorization) => {
+  const {
+    experimentId: parentExperimentId,
+    subsetExperimentId,
+  } = params;
+
+  const {
+    processingConfig: parentProcessingConfig,
+  } = await new Experiment().findById(parentExperimentId).first();
+
+  const { stateMachineArn, executionArn } = await createSubsetPipeline(
```

Comment:
It feels a bit weird that `createSubsetPipeline` is called after `executeSubsetPipeline`. 

I can't think of a good alternative for either of these names though, do you think it is worth it to rename it to `createSubsetStateMachine`? Maybe it shoudl just be kept like this until we refactor `createSubsetPipeline` and its counterparts for each pipe type